hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -1026,22 +1107,25 @@ private void clean(String startCleanTime) throws HoodieIOException {
 
       // Create the metadata and save it
       HoodieCleanMetadata metadata = AvroUtils
-          .convertCleanMetadata(startCleanTime, durationInMs, cleanStats);
+          .convertCleanMetadata(cleanInstant.getTimestamp(), durationInMs, cleanStats);
       logger.info("Cleaned " + metadata.getTotalFilesDeleted() + " files");
-      metrics
-          .updateCleanMetrics(durationInMs.orElseGet(() -> -1L), metadata.getTotalFilesDeleted());
+      metrics.updateCleanMetrics(durationInMs.orElseGet(() -> -1L), metadata.getTotalFilesDeleted());
 
       table.getActiveTimeline()
-          .saveAsComplete(new HoodieInstant(true, HoodieTimeline.CLEAN_ACTION, startCleanTime),
-              AvroUtils.serializeCleanMetadata(metadata));
-      logger.info("Marked clean started on " + startCleanTime + " as complete");
+          .transitionCleanInflightToComplete(new HoodieInstant(true, HoodieTimeline.CLEAN_ACTION,
+                  cleanInstant.getTimestamp()), AvroUtils.serializeCleanMetadata(metadata));
+      logger.info("Marked clean started on " + cleanInstant.getTimestamp() + " as complete");
 
+      /**
+       * @vinoth/@n3nash : I have commented this code as archiving can handle cleaner actions. Is this ok ?
       if (!table.getActiveTimeline().getCleanerTimeline().empty()) {
         // Cleanup of older cleaner meta files
         // TODO - make the commit archival generic and archive clean metadata
         FSUtils.deleteOlderCleanMetaFiles(fs, table.getMetaClient().getMetaPath(),
             table.getActiveTimeline().getCleanerTimeline().getInstants());
       }
+      **/
```

Comment:
Should this snippet of code removed?

Reply:
Will confirm with @n3nash  before I remove this part.